packages/connect/src/impl/dynamic.ts

```diff
@@ -0,0 +1,134 @@
+import EventEmitter from 'events';
+
+import { ConnectFactoryDependencies } from '../factory';
+import type { ConnectSettingsPublic, Manifest } from '../types';
+import { CallMethodPayload } from '../events';
+import { ERRORS } from '../constants';
+
+import { ProxyEventEmitter } from '../utils/proxy-event-emitter';
+
+type TrezorConnectDynamicParams<TrezorConnectImplType> = {
+    implementations: {
+        type: TrezorConnectImplType;
+        impl: ConnectFactoryDependencies;
+    }[];
+    getInitTarget: (settings: Partial<ConnectSettingsPublic>) => TrezorConnectImplType;
+    handleErrorFallback: (errorCode: string) => Promise<boolean>;
+};
+
+/**
+ * Implementation of TrezorConnect that can dynamically switch between different implementations.
+ *
+ */
+export class TrezorConnectDynamic<TrezorConnectImplType> implements ConnectFactoryDependencies {
+    public eventEmitter: EventEmitter;
+
+    private currentTarget: TrezorConnectImplType;
+    private implementations: TrezorConnectDynamicParams<TrezorConnectImplType>['implementations'];
+    private getInitTarget: TrezorConnectDynamicParams<TrezorConnectImplType>['getInitTarget'];
+    private handleErrorFallback: TrezorConnectDynamicParams<TrezorConnectImplType>['handleErrorFallback'];
+
+    public lastSettings?: { manifest: Manifest } & Partial<ConnectSettingsPublic>;
+
+    public constructor({
+        implementations,
+        getInitTarget,
+        handleErrorFallback,
+    }: TrezorConnectDynamicParams<TrezorConnectImplType>) {
+        this.implementations = implementations;
+        this.currentTarget = this.implementations[0].type;
+        this.getInitTarget = getInitTarget;
+        this.handleErrorFallback = handleErrorFallback;
+        this.eventEmitter = new ProxyEventEmitter(
+            this.implementations.map(impl => impl.impl.eventEmitter),
+        );
+    }
+
+    private getTarget() {
+        return this.implementations.find(impl => impl.type === this.currentTarget)!.impl;
+    }
+
+    public async switchTarget(target: TrezorConnectImplType) {
+        if (this.currentTarget === target) {
+            return;
+        }
+
+        if (!this.lastSettings) {
+            throw new Error('Connect dynamic not initialized.');
+        }
+        this.currentTarget = target;
```

Comment:
Previously there was also dispose before switching
```
await this.getTarget().dispose();
```

Reply:
Right, that was the breaking CI thing.